src/main/java/io/vavr/collection/Iterator.java

```diff
@@ -1211,6 +1211,20 @@ public U next() {
         }
     }
 
+    @Override
+    default <T1, T2> Tuple2<Iterator<T1>, Iterator<T2>> unzip(
+            Function<? super T, ? extends T1> unzipper1, Function<? super T, ? extends T2> unzipper2) {
+        // return unzip(i -> Tuple.of(unzipper1.apply(i), unzipper2.apply(i)));
+        Objects.requireNonNull(unzipper1, "unzipper1 is null");
+        Objects.requireNonNull(unzipper2, "unzipper2 is null");
+        if (!hasNext()) {
+            return Tuple.of(empty(), empty());
+        } else {
+            final Stream<Tuple2<? extends T1, ? extends T2>> source = Stream.ofAll(() -> map(i -> Tuple.of(unzipper1.apply(i), unzipper2.apply(i))));
```

Comment:
This one looks particularly questionable, as we do create a Stream of Tuples here, and then split them into Tuple of Iterators. However I could not find a better solution yet, as Iterator has mutable state, and generally traversable only once.

Reply:
Are there better ways to do this?

Reply:
Right, I would do it as follows:

```java
// ...
final Iterator<T1> iter1 = map(unzipper1);
final Iterator<T2> iter2 = map(unzipper2);          
return Tuple.of(iter1, iter2);
```